doc/vrrp/VRRP_Adaptation_HLD.md

```diff
@@ -0,0 +1,712 @@
+
+![VRRP Tracking Interface Scenarios](images/VRRP_Tracking_Interface_Scenarios_Diagram.png "Figure 1: VRRP Tracking Interface Scenarios")
+
+As shown in the figure，Configure VRRP backup groups between RouterA and RouterB, Both RouterA and RouterB operate in preemptive mode. On RouterA, configure to monitor the uplink interface Interface1. When Interface1 fails, RouterA lowers its priority and, through packet negotiation, RouterB preempts and becomes the Master, ensuring that user traffic is forwarded normally.
+
+### Architecture Design 
+
+Below diagram illuminates the interactions between relevant components and the DB involved for VRRP support in SONiC architecture at a high level.
+
+![VRRP Architecture](images/VRRP_Architecture_Diagram.png "Figure 2: VRRP in SONiC Architecture")
+
+Diagram 1. VRRP components.
+
+Significantly, This feature extends functionality implemented in [SONiC FRR-BGP Extended Unified Configuration Management Framework](https://github.com/sonic-net/SONiC/blob/master/doc/mgmt/SONiC_Design_Doc_Unified_FRR_Mgmt_Interface.md), Redis DB events will trigger frrcfgd when the field frr_mgmt_framework_config set to "true" in the DEVICE_METADATA table, otherwise will trigger vrrpcfgd, and then frrcfgd or vrrpcfgd will configure FRR-VRRP using FRR CLI commands.
+
+### High-Level Design
+
+#### Operating environment
+
+Since support for placing macvlan devices into protodown was not added to Linux until version 5.1, the kernel version is required:
+- Linux kernel version 5.1+
+
+#### Scalability Requirements
+
+- Max number of VRRP instances: 254
+
+#### Container
+
+##### BGP container
+
+vrrpcfgd: 
+  - Subscribes to CONFIG_DB tables, parsing configurations and passes to vrrpd by using FRR CLI commands. 
+
+vrrpd: 
+  - Responsible for all VRRP protocol related calculations. VRRP packets are sent and received in vrrpd and states are updated accordingly.
+
+vrrpmgrd: 
+  - Listens to VRRP create, delete and parameter change in CONFIG_DB. Complete the following tasks:
+    - Add/del Linux Macvlan device to kernel
+    - Config virtual MAC to Macvlan device;
+    - Add/del VIP to Macvlan device;
+    - Upadate VRRP instance configuration to the APPL DB.
+
+vrrpsyncd: 
+  - Complete the following tasks:
+    - Listen for kernel Macvlan device and other interfaces protocol status change events
+    - Update the kernel Macvlan device status to the APPL DB.
+    - Match VRRP instance tracking interface and recalculate priority.
+    - Update priority change to vrrpd by using FRR CLI commands.
+
+##### SWSS container
+
+vrrporch: 
+  - Subscribes to APPL_DB tables, responsible for updating the ASIC DB. Creating/deleting Virtual RIF object via SAI API.
```

Comment:
I dont see a need for separate vrrporch and needs to only handle few extra attributes for INTERFACE creation. Please consider to leverage intforch

Reply:
OK